gnome3: 3.28 → 3.30

jtojnar opened the pull request:
###### Motivation for this change
Requires [Meson 0.47.0](https://github.com/NixOS/nixpkgs/pull/46020), it will need to be fixed first.

https://hydra.nixos.org/jobset/nixpkgs/gnome

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

jtojnar committed:
gnome-recipes: init at 2.0.2

jtojnar committed:
libhandy: init at 0.0.5

worldofpeace commented:
@jtojnar This is problematic for me because [gala](https://github.com/elementary/gala/issues/303) will fail to compile with this version of mutter. Is there some nix'y way I can work around this until they have a fix?

jtojnar commented:
@worldofpeace  You could re-add the old mutter expression as is done with VTE for pantheon-terminal, for example.

worldofpeace commented:
Mind just including that in nixpkgs here? I don't know when they'll get around to fixing it, and it's already a lot to build.

jtojnar commented:
Taking a note.

jtojnar committed:
vala_0_42: init at 0.42.3

jtojnar committed:
webkitgtk: 2.20.5 → 2.22.0

jtojnar committed:
telepathy-glib: clean up

jtojnar committed:
vala: enable parallel building

jtojnar committed:
libpsl: 0.17.0 → 0.20.2

jtojnar committed:
gnome3.adwaita-icon-theme: 3.28.0 → 3.30.1

jtojnar committed:
gnome3.aisleriot: 3.22.5 → 3.22.7

jtojnar committed:
atk: 2.28.1 → 2.30.0

jtojnar committed:
baobab: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-notes: 3.28.3 → 3.30.3

jtojnar committed:
gnome3.cheese: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.dconf: 0.28.0 → 0.30.1

jtojnar committed:
gnome3.dconf-editor: 3.28.0 → 3.30.2

jtojnar committed:
epiphany: 3.28.3.1 → 3.30.2

jtojnar committed:
evince: 3.28.2 → 3.30.2

jtojnar committed:
gnome3.evolution: 3.28.5 → 3.30.2

jtojnar committed:
evolution-data-server: 3.28.5 → 3.30.2

jtojnar committed:
gnome3.file-roller: 3.28.1 → 3.30.1

jtojnar committed:
gnome3.five-or-more: 3.28.0 → 3.30.0

jtojnar committed:
gdk_pixbuf: 2.36.12 → 2.38.0

jtojnar committed:
gnome3.gdm: 3.28.3 → 3.30.2

jtojnar committed:
gnome3.gedit: 3.28.1 → 3.30.2

jtojnar committed:
gnome3.gjs: 1.52.3 → 1.54.3

jtojnar committed:
glib: 2.56.0 → 2.58.1

jtojnar committed:
glib-networking: 2.56.0 → 2.58.0

jtojnar committed:
gnome3.gnome-backgrounds: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-boxes: 3.28.5 → 3.30.3

jtojnar committed:
gnome3.gnome-calculator: 3.28.2 → 3.30.1

jtojnar committed:
gnome3.gnome-calendar: 3.28.2 → 3.30.1

jtojnar committed:
gnome3.gnome-characters: 3.28.2 → 3.30.0

jtojnar committed:
gnome3.gnome-chess: 3.28.1 → 3.30.0

jtojnar committed:
gnome3.gnome-clocks: 3.28.0 → 3.30.1

jtojnar committed:
gnome3.gnome-color-manager: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-contacts: 3.28.2 → 3.30.1

jtojnar committed:
gnome3.gnome-control-center: 3.28.2 → 3.30.2

jtojnar committed:
gnome3.gnome-desktop: 3.28.2 → 3.30.2

jtojnar committed:
gnome3.gnome-disk-utility: 3.28.3 → 3.30.2

jtojnar committed:
gnome3.gnome-documents: 3.28.2 → 3.30.0

jtojnar committed:
gnome3.gnome-logs: 3.28.5 → 3.30.0

jtojnar committed:
gnome3.gnome-maps: 3.28.2 → 3.30.2.1

jtojnar committed:
gnome3.gnome-mines: 3.28.0 → 3.30.1.1

jtojnar committed:
gnome3.gnome-music: 3.28.2.1 → 3.30.2

jtojnar committed:
gnome3.gnome-online-accounts: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-online-miners: 3.26.0 → 3.30.0

jtojnar committed:
gnome3.gnome-packagekit: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-photos: 3.28.0 → 3.30.1

jtojnar committed:
gnome3.gnome-power-manager: 3.26.0 → 3.30.0

jtojnar committed:
gnome3.gnome-screenshot: 3.26.0 → 3.30.0

jtojnar committed:
gnome3.gnome-session: 3.28.1 → 3.30.1

jtojnar committed:
gnome3.gnome-settings-daemon: 3.28.1 → 3.30.1.2

jtojnar committed:
gnome3.gnome-shell: 3.28.3 → 3.30.2

jtojnar committed:
gnome3.gnome-shell-extensions: 3.28.1 → 3.30.1

jtojnar committed:
gnome3.gnome-software: 3.28.2 → 3.30.5

jtojnar committed:
gnome3.gnome-sudoku: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-system-monitor: 3.28.2 → 3.30.0

jtojnar committed:
gnome3.gnome-taquin: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.gnome-terminal: 3.28.2 → 3.30.2

jtojnar committed:
gnome3.gnome-tweaks: 3.28.1 → 3.30.1

jtojnar committed:
gobjectIntrospection: 1.56.0 → 1.58.1

Upstream now strips absolute paths to their basename on all platforms apart from
Darwin: https://gitlab.gnome.org/GNOME/gobject-introspection/commit/a41abe1868a693387cd5cf85567cf2e0fd6c62df

To get around this without modifying the basename test we simply pass in
`basename=False` when normally generating gir files.

jtojnar committed:
gsettings-desktop-schemas: 3.28.0 → 3.28.1

jtojnar committed:
gnome3.gtksourceview: 3.24.6 → 3.24.8

jtojnar committed:
gucharmap: 11.0.1 → 11.0.3

jtojnar committed:
gvfs: 1.36.2 → 1.38.1

jtojnar committed:
gnome3.iagno: 3.28.0 → 3.30.0

jtojnar committed:
libdazzle: 3.28.5 → 3.30.2

jtojnar committed:
librsvg: 2.42.4 → 2.44.9

jtojnar committed:
libsecret: 0.18.5 → 0.18.6

jtojnar committed:
libsoup: 2.62.2 → 2.64.2

jtojnar committed:
gnome3.lightsoff: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.metacity: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.mutter: 3.28.3 → 3.30.2

jtojnar committed:
gnome3.nautilus: 3.28.1 → 3.30.4

jtojnar committed:
orca: 3.28.2 → 3.30.1

jtojnar committed:
gnome3.polari: 3.28.1 → 3.30.2

jtojnar committed:
gnome3.seahorse: 3.20.0 → 3.30

jtojnar committed:
shotwell: 0.28.2 → 0.30.0

jtojnar committed:
simple-scan: 3.28.1 → 3.30.2

jtojnar committed:
gnome3.sushi: 3.28.3 → 3.30.0

jtojnar committed:
gnome3.swell-foop: 3.28.0 → 3.30.0

jtojnar committed:
gnome3.tracker-miners: 2.1.3 → 2.1.5

jtojnar committed:
gnome3.vte: 0.52.2 → 0.54.2

jtojnar committed:
yelp: 3.28.1 → 3.30.0

jtojnar committed:
gnome3.yelp-xsl: 3.28.0 → 3.30.1

jtojnar committed:
gtk3: 3.22.30 → 3.24.1

hedning commented:
Should we just add the meson upgrade here? Then we can debug it with hydra.

jtojnar commented:
Yeah, I did that, just was pushing to a wrong branch.

jtojnar committed:
gnome3.gconf: remove

The throw statement broke building gnome3 attribute set. Since 18.09
has already been branched off, we just remove the warning.

jtojnar committed:
libhttpseverywhere: fix build

hedning commented:
pushed some fixes to https://github.com/hedning/nixpkgs/commits/gnome-3.30, marked with fixup! or squash! where appropriate (so it's easy to squash things down).  

- pipewire to 0.2.2, I had this laying around, 0.2.3 didn't build without any effort
- epiphany needs libdazzle now. It does build, but fails one test
- gtksourceview should be updated to 3.24.8

hedning committed:
spidermonkey_60: init at 60.3.0

A new test checks and fails if we're building in js/src, so we create a build
dir like Arch does.

hedning committed:
sysprof: 3.28.1 → 3.30.0

hedning commented:
Some more fixes at https://github.com/hedning/nixpkgs/commits/gnome-3.30:

- `xdg-desktop-portal`: fails due to new pipewire, updated this and flatpak and related packages
- `evince` now depends on gspell
- `sysprof` is fixed by updating to 3.30
- `gnome-system-monitor` require meson
- `libgda` has patch upstream now

Some notes on other failures:

- `tracker`: libsoup upgrade is the cause here it seems, build is fixed by reverting (might be good to do as there's a lot of packages that depend on tracker).
- `libunique` fails on a warning
- `libdazzle` fails on a test as noted above 
- `gjs`: we need 60 now obviously
- `gnome-online-accounts` fails to produce the `man` output (release notes also mentions telepathy no longer being supported) not sure what the problem is

I'm stuck on spidermonkey [here](https://github.com/hedning/nixpkgs/commit/808c33ce17c5cc5e2fec9642bdbb6065ea050479), getting this error in the configure phase:

```
Traceback (most recent call last):
  File "./../../configure.py", line 127, in <module>
    sys.exit(main(sys.argv))
  File "./../../configure.py", line 34, in main
    return config_status(config)
  File "./../../configure.py", line 122, in config_status
    return config_status(args=[], **encode(sanitized_config, encoding))
  File "/build/firefox-60.2.0/python/mozbuild/mozbuild/config_status.py", line 144, in config_status
    definitions = list(definitions)
  File "/build/firefox-60.2.0/python/mozbuild/mozbuild/frontend/emitter.py", line 184, in emit
    objs = list(emitfn(out))
  File "/build/firefox-60.2.0/python/mozbuild/mozbuild/frontend/emitter.py", line 1134, in emit_from_context
    full_path), context)
mozbuild.frontend.reader.SandboxValidationError:
==============================
FATAL ERROR PROCESSING MOZBUILD FILE
==============================

The error occurred while processing the following file or one of the files it includes:

    /build/firefox-60.2.0/js/src/shell/moz.build

The error occurred when validating the result of the execution. The reported error is:

    Path specified in LOCAL_INCLUDES is not allowed: .. (resolved to /build/firefox-60.2.0/js/src)
```

jtojnar commented:
I suspect the tracker failure is caused by the meson rpath issue (cannot reproduce with libsoup built with autotools).

IIRC libdazzle tests are non-deterministic, I should have disabled them long time ago. **Done.**

Looking at the GOA diff, I see no reason for man pages stopping being built, **fixed**.

Reading https://git.archlinux.org/svntogit/packages.git/tree/repos/gnome-unstable-x86_64/PKGBUILD?h=packages/js60, I tried dropping the autoconf and building from a different directory but it did not help. Maybe ask at irc://irc.mozilla.org/jsapi.

hedning commented:
> I suspect the tracker failure is caused by the meson rpath issue (cannot reproduce with libsoup built with autotools).

Right, any idea how to debug this? Doing a `readelf -d` doesn't seem to produce anything weird at least (freenode is being weird and requiring registered nicks, which doesn't seem to work with matrix...).

jtojnar commented:
I am not really sure. I am still going through the meson commits.

jtojnar committed:
gnome3.atomix: 3.29.3 → 3.30.0.1

jtojnar committed:
gnome3.gnome-getting-started-docs: 3.28.2 → 3.30.0

jtojnar committed:
gnome3.gnome-user-docs: 3.28.2 → 3.30.2

hedning commented:
Looked at spidermonkey some more. The test that's failing is new in 60, commenting it out «works». Running configure from another directory also works, which is what Arch do.

hedning commented:
Pushed working spidermonkey to https://github.com/hedning/nixpkgs/commits/gnome-3.30

`gjs` fails due some path weirdness where it tries to link to a file starting with `.//nix/store`:
```
config.status: linking /nix/store/gjxbz4mbyd67vfia9gljwwiiyj6lm4na-gobject-introspection-1.58.0-dev/share/gobject-introspection-1.0/tests/gitestmacros.h to installed-tests/gitestmacros.h
config.status: linking /nix/store/gjxbz4mbyd67vfia9gljwwiiyj6lm4na-gobject-introspection-1.58.0-dev/share/gobject-introspection-1.0/tests/regress.c to installed-tests/regress.c
config.status: linking /nix/store/gjxbz4mbyd67vfia9gljwwiiyj6lm4na-gobject-introspection-1.58.0-dev/share/gobject-introspection-1.0/tests/regress.h to installed-tests/regress.h
config.status: linking .//nix/store/gjxbz4mbyd67vfia9gljwwiiyj6lm4na-gobject-introspection-1.58.0-dev/share/gobject-introspection-1.0/tests/warnlib.c to installed-tests/warnlib.c
config.status: error: .//nix/store/gjxbz4mbyd67vfia9gljwwiiyj6lm4na-gobject-introspection-1.58.0-dev/share/gobject-introspection-1.0/tests/warnlib.c: file not found
```

hedning commented:
Pushed some more fixes:

- gdm: needed a patch to make udev install dir configurable
- seahorse: requires meson, didn't figure out pkcs11 support
- gucharmap: patch is upstream
- gnome-system-monitor: cleanup
- evolution-data-server: complained about libcanberra and pcre (haven't tested build yet, webkitgtk was blocking)

hedning commented:
webkitgtk failed due to a transient hardware error (not enough space left on device), so the job needs to be restarted :( Really wish there was some automation for these things.

hedning committed:
nixos/gdm: use XDG_DATA_DIRS to find sessions

Gdm now searches for session files in XDG_DATA_DIRS so we no longer need the
sessions_dir.patch.

jtojnar committed:
pitivi: 0.99.0 → 0.999.0

jtojnar commented:
Seahorse with pkcs11 works for me.

Apparently, meson’s `absolute-shlib.patch` no longer works (e.g. `atk`), it still works with GNUMake (`gnome3.rest`).

FRidh commented on pkgs/development/python-modules/gst-python/default.nix:
```diff
@@ -1,10 +1,10 @@
-{ buildPythonPackage, fetchurl, stdenv, meson, ninja, pkgconfig, python, pygobject3
+{ buildPythonPackage, fetchurl, stdenv, meson, ninja, pkgconfig, python, python3, pygobject3
```
The `python` here is the Python used for the package set. Does the `python2` build also require `python3`?

jtojnar replied:
Yes, it does.

jtojnar replied:
Actually, it looks like it only supports Python 3 now: https://github.com/GStreamer/gst-python/blob/master/meson.build#L22

hedning commented:
Looks like [this is the problem](https://gitlab.gnome.org/GNOME/gobject-introspection/commit/a41abe1868a693387cd5cf85567cf2e0fd6c62df): 

```python
    if basename:
        shlibs = list(map(os.path.basename, shlibs))
    return shlibs
```

`basename` is true for everything that isn't darwin.

hedning commented:
No idea why things work with make though.

hedning commented:
Looking at libsoup [upstream there's a few fixes for meson which aren't in 2.64](https://gitlab.gnome.org/GNOME/libsoup/commits/master). In particular the issue with tracker (an extra `_`) should probably be fixed by this: https://gitlab.gnome.org/GNOME/libsoup/commit/f134b372d34131d94a6aadd44f351e52a005f8d6

hedning commented:
Pushed a fix libsoup with patches applied from upstream. Just took them all apart from the `gmtime_r` as it caused the build to fail due to a warning.

hedning commented:
Huh, did `nix-prefetch-url` on the patches and built locally, but I'm guessing `nix-prefetch-url` gives different sha then fetchpatch (and since I had prefetched the local build worked).

jtojnar commented:
Yeah, `fetchpatch` produces a normalized patch.

hedning committed:
pythonPackages.pygobject3: 3.26.1 -> 3.30.2

hedning committed:
gnome3.gnome-panel: 3.28.0 → 3.30.0

hedning commented:
Some more fixes at https://github.com/hedning/nixpkgs/commits/gnome-3.30:

- gnome-shell failed due to obsolete `--sysconfdir=/etc` flag
- nautilus now depends on libseccomp
- gnome-music now depends on libdazzle, libsoup and pygobject > 3.29.1
- upgraded gnome-panel to 3.30.0

hedning committed:
gnome3.geary: fix build with webkit-2.22.0

hedning committed:
gnome3.gnome-applets: 3.28.0 → 3.30.0

hedning committed:
gnome3.gnome-flashback: 3.28.0 → 3.30.0

hedning commented:
Pushed some updates and a fix to build geary with the new webkitgtk.

Evolution tries to install things in evolution-data-server now, coming from this place: https://gitlab.gnome.org/GNOME/evolution/blob/master/src/modules/alarm-notify/CMakeLists.txt#L1. Doesn't seem to be configurable, so we'll probably need to do some patching.

hedning committed:
gnome3.gnome-usage: 3.28.0 → 3.30.0

hedning committed:
libunique: Don't fail on deprecated-declarations

hedning commented:
Some more fixes:
- updated the flatpak patches so it builds now (haven't run the test)
- gnome-usage: added an unmerged patch that fixed the build with vala_42
- lightsoff: switched to meson
- `gnome-calendar` moved `meson_post_install.py`
- `libunique`: don't error on deprecrated-declarations

Remaining breakage:
- `gnome-photos` can't find gegl for some reason
- `folks` can't find `gee:
```
checking for gio-2.0 gee-0.8 vala bindings... configure: error: Package requirements were not met: gio-2.0
                             gee-0.8

error: Package `gee-0.8' not found in specified Vala API directories or GObject-Introspection GIR directories
```
- `gnome-panel` fails due to `clock-applet.c:44:10: fatal error: gio/gdesktopappinfo.h: No such file or directory`
- `anjuta` fails to find libgda for some reason
- evolution as mentioned above

jtojnar committed:
gnome3.folks: fix build

jtojnar commented:
Still building but hopefully these fixes should do it:

* gnome-photos needs newer gegl
* folks needed gobject-introspection setup hook
* gnome-panel suffers from https://gitlab.gnome.org/GNOME/gnome-panel/issues/8
* evolution needs pkg-config variable override
* we have updated libgda to a pre-release version of 6.0 but anjuta requires even newer one that does not exist (versioning scheme is a mess). We should probably pin libgda to 5.0 for now.

---

* anjuta also requires Vala 0.40, we might need to patch it with https://gitlab.gnome.org/GNOME/anjuta/commit/3d2cb92c857b1288aca7d96886cee03714a5f0ff

hedning commented:
`folks` isn't passed `gobjectIntrospection`

jtojnar committed:
gnome3.gpaste: 3.28.2 → 3.30.2

hedning commented:
Ah, managed to not check the python-2 version of pygobject it seems, so that broke a bunch a things.

hedning commented:
Pushed a fix for pygobject, not sure if there's a better way to easily detect python version though.

jtojnar committed:
zeitgeist: fix build

jtojnar committed:
gnome3.gnome-devel-docs: 3.28.0 → 3.30.2

jtojnar committed:
gnome3.pomodoro: fix build

jtojnar committed:
gnome-builder: 3.28.4 → 3.30.0

jtojnar committed:
jsonrpc-glib: 3.28.1 → 3.30.0

jtojnar committed:
at-spi2-atk: 2.26.2 → 2.30.0

jtojnar committed:
at-spi2-core: 2.28.0 → 2.30.0

jtojnar commented:
I went through all the NEWS files; the code should be in reasonable state. Now he following remains:

- [ ] update shell extensions
- [ ] change gst-python back to autotools for python 2 support
- [ ] build devdoc for glib
- [ ] re-add mutter-3.0 for elementary
- [ ] fix meson rpath patch
- [ ] rebase

jtojnar committed:
gnomeExtensions.dash-to-dock: 63 → 64

jtojnar committed:
gnomeExtensions.dash-to-panel: 11 → 16

hedning commented:
Pushed a change to pygobject3 that uses `isPy3k` instead of substring on python version. Will cause some rebuilds, but I 'm guessing that's the preferred way of detecting what python one is dealing with.

hedning committed:
gnomeExtensions.icon-hider: 19 -> 20

hedning committed:
gnomeExtensions.nohotcorner: 16.0 -> 17.0

hedning committed:
gnomeExtensions.system-monitor: 33 -> 36

jtojnar committed:
gnome-latex: 3.30.1 → 3.30.2

jtojnar committed:
gtksourceview4: 4.0.0 → 4.0.3

hedning commented:
Note I only updated shell extensions with new releases, so there's a few I think that probably should be updated to a new rev.

jtojnar commented on pkgs/desktops/gnome-3/core/gnome-shell/default.nix:
```diff
@@ -13,16 +13,13 @@ let
   };
 
-  # Needed to find /etc/NetworkManager/VPN
-  mesonFlags = [ "--sysconfdir=/etc" ];
```
I wonder why was this removed?

hedning replied:
The build tried to install autostart files there. The networkmanager code is gone from upstream.

hedning replied:
Left something in the commit message, but should've marked it as squash. The relevant bit I guess:
> The flag should no longer be needed for networkmanager detection as the code
from 3.22 is no longer there (js/ui/components/networkAgent.js:768 @3.22.0).

jtojnar commented on pkgs/development/libraries/flatpak/fix-test-paths.patch:
```diff
@@ -9,16 +9,7 @@
          sed -e 's/^/# /' < bwrap-result
          echo "1..0 # SKIP Cannot run bwrap"
          exit 0
-@@ -323,7 +323,7 @@
```
Why were these fragments removed?

hedning replied:
The patches didn't apply, in particular the `skip_without_python2` function is gone. Probably should've left an explanation in the commit message. Looks like there's no python2 support anymore.

jtojnar commented:
So did I. I will leave the other extensions to their maintainers.

hedning commented on pkgs/development/libraries/flatpak/fix-test-paths.patch:
```diff
@@ -35,21 +26,18 @@
-     cp `which ldconfig` ${DIR}/usr/bin
- fi
--LIBS=`mktemp`
--BINS=`mktemp`
```
Hmm, this probably shouldn't have been deleted though, when I look at it again. Not a big deal, but they're unused so.

hedning replied:
Pushed commit to add back these deletions.

jtojnar committed:
empathy: fix build

jtojnar commented:
I rebased the flatpak patch onto 1.0.2.

hedning commented:
Okay, now for some runtime errors :laughing:

gnome-session fails to find `gio-launch-desktop` from `glib.dev`. Wrapping gnome-session with `${glib.dev}/bin` lets us launch gdm and then gnome-shell.

It looks like it's glib itself that calls `gio-launch-desktop`: https://gitlab.gnome.org/GNOME/glib/blob/master/gio/gdesktopappinfo.c#L2728 (running `rg gio-launch-desktop --text` on the closure of the VM returned only the .../lib/libgio-2.0.so.0.5800.0). So that's probably the correct place to patch as Glib should know how to refer to itself.

https://github.com/NixOS/nixpkgs/pull/38486 have landed in staging, but `gio-launch-desktop` is still in `dev` so that shouldn't really affect us.

hedning commented:
Is there a better way to do this, I ran into problems using substituteAll:
```diff
modified   pkgs/development/libraries/glib/default.nix
@@ -59,7 +59,10 @@ stdenv.mkDerivation rec {
     ++ optionals stdenv.hostPlatform.isMusl [
       ./quark_init_on_demand.patch
       ./gobject_init_on_demand.patch
-    ] ++ [ ./schema-override-variable.patch ];
+    ] ++ [
+      ./schema-override-variable.patch
+      ./fix-gio-launch-desktop-path.patch
+    ];
 
   outputs = [ "out" "dev" ];
   outputBin = "dev";
@@ -87,6 +90,7 @@ stdenv.mkDerivation rec {
     patchShebangs glib/gen-unicode-tables.pl
     patchShebangs tests/gen-casefold-txt.py
     patchShebangs tests/gen-casemap-txt.py
+    substituteInPlace gio/gdesktopappinfo.c --replace "@outputBin@" $dev
   '';
 
   LIBELF_CFLAGS = optional stdenv.isFreeBSD "-I${libelf}";
new file   pkgs/development/libraries/glib/fix-gio-launch-desktop-path.patch
@@ -0,0 +1,11 @@
+--- a/gio/gdesktopappinfo.c
++++ b/gio/gdesktopappinfo.c
+@@ -2725,7 +2725,7 @@ g_desktop_app_info_launch_uris_with_spawn (GDesktopAppInfo            *info,
+ 
+           /* Fall back on usual searching in $PATH */
+           if (tmp == NULL)
+-            tmp = "gio-launch-desktop";
++            tmp = "@outputBin@/bin/gio-launch-desktop";
+           g_once_init_leave (&gio_launch_desktop_path, tmp);
+         }
+ 
```

jtojnar commented:
Unfortunately not, due to https://github.com/NixOS/nix/issues/1846

hedning committed:
glib: use absolute path for `gio-launch-desktop`

glib tries to look up `gio-launch-desktop` in the path which obviously can fail.
Specifically this now happens when launching gnome-shell.

hedning commented:
Hmm, any idea how to avoid, I'm guessing an output can refer to itself, but not to another output that refers back to it?
```
cycle detected in the references of '/nix/store/lbzqfdj8rq9dm62867fxwycjaxv4sk4s-glib-2.58.0-dev' from '/nix/store/rsrqz77vghpgykzmf17jg95fgh1qxxlv-glib-2.58.0'
```

jtojnar commented:
Would moving the file to `bin` help? It should most likely go there anyway. Though I doubt it, the `bin` will most likely depend on the `!outputLib` (`out`) anyway.

Maybe move this one to out and add a symlink to it from bin if it is expected in the environment.

hedning commented:
Referring to `out` worked, though `bin` might be more appropriate. We'd need to grab https://github.com/NixOS/nixpkgs/pull/38486 first though.

hedning commented:
Pushed something that probably should work for glib: https://github.com/hedning/nixpkgs/commit/4b754a2db253a2c974675fff50cc03c2975ec58a

Though we'll might want to rework it after staging-next hits master (where `glib` has a `bin` output).

hedning commented:
So the recursion happens when going down to `dblatex` which isn't required to build `gtk-doc`. So the cleanest fix is probably to pass in a `gtk-doc` without `dblatex`.

hedning commented:
Using a `gtk-doc` without `dblatex` for `glib` makes things build, but something fails when it tries to install the docs, looks like it's in meson's `gnome.gtkdoc` function, but there's no useful error message, just `FAILED: meson-install`, so not sure what the problem is.

jtojnar committed:
template-glib: 3.28.0 → 3.30.0

jtojnar committed:
python3.pkgs.pyatspi: 2.26.0 → 2.30.0

hedning committed:
playerctl: switch to meson and fix build

hedning commented:
This is a pretty strange failure, which I'm able to reproduce: [playerctl](https://hydra.nixos.org/build/81609167/log) the path it's complaining about exists, so 

Switching to meson fixed it, though I had to turn off docs as it looks for a file in the wrong place.

worldofpeace committed:
gnome3.mutter328: init

hedning commented:
Things are looking good. Running this branch now (with wayland commits on top). The first reboot was weird though, GDM spawned on tty2 and didn't pick up all the session files (specifically didn't show `GNOME on Xorg`), it also launched Xorg instead of wayland. I'm unable to reproduce this so :man_shrugging: .

Are we sure meson is still a problem? We haven't really found any problems with it yet right? The version that was reverted was 0.47.0 and we're using 0.47.2 now, so it might be fixed upstream. If so I'm guessing https://github.com/mesonbuild/meson/commit/833e3e7d8d6e513525b0cc1debb376fc1d83a908 was the fix. cc @dtzWill 

The hydra job compared against the correct trunk seems to mostly have transient errors (I'm guessing the tarball fails due to the temporary update hack): https://hydra.nixos.org/eval/1479257?compare=1478835

hedning committed:
xfce.xfce4-namebar-plugin: Specify vala_0_40

hedning committed:
telepathy-salut: use standard pkgconfig

Building glib with selinux support caused the build to fail due to selinux being
specificed in `Required.private` in gio-2.0.pc. `pkgconfigUpstream` isn't
patched to handle `Required.private` properly.

hedning committed:
planner: don't error on deprecrated declarations

hedning committed:
ldmtool: don't fail on deprecated declarations

hedning committed:
gnucash24: add `pcre` fix build

hedning commented:
Yeah, there's few failures left: pushed a few fixes to https://github.com/hedning/nixpkgs/tree/gnome-3.30. I just added `libselinux` and `libsepol` (which it also complained about) to `telepathy-salut`. But I guess it might indicate something wrong with glib as building with selinux is new for us. Nothing enlightening in the NEWS file though.

Apart from that I found a run time bug in gnome-shell today. When trying to connect to a network, clicking the `Connect` button doesn't do anything, Seems like it's an upstream bug: https://gitlab.gnome.org/GNOME/gnome-shell/commit/33ffdd60611e8e8d0018680dd97bcbd0e98c6b4b so we should probably pick that patch.

hedning commented:
`libselinux` is specified as `Requires.private` in the `gio-2.0.pc` file, so I assume we shouldn't propagate it. 

Looking a bit more closely at `telepathy-salut` I noticed that it uses `pkgconfigUpstream`, replacing it with the default `pkgconfig` fixed the build. `pkconfigUpstream` has `vanilla` passed as true, so doesn't handle `Required.private` properly: https://github.com/NixOS/nixpkgs/blob/8b58a7187c9f3c8f1997f328c27c536effccdac3/pkgs/development/tools/misc/pkgconfig/default.nix#L17-L20 

So the correct fix is to use the default `pkgconfig` here. `pkgconfigUpstream` was only used in `telepathy-salut` which means we should perhaps just drop it?

I pushed the fix to my branch.

dtzWill commented:
Hmm well if the problem appears solved I'll stand down and here's hoping we never have to worry about it.  :D.

Keep it in mind if things seem crazy later but otherwise let's hope it's fixed and we never have to worry about it again :).  Thanks for being careful, sorry for not having more time to investigate myself.

hedning commented:
This should be all new failures when compared to job 1478835:

### Remaining failures
- [`gnucash`](https://hydra.nixos.org/build/81615471/log) deprecrated-declaration from glib, but the regular fix doeesn't work
- [`gxneur`](https://hydra.nixos.org/build/81606559/log) glib deprecated usage, has patch upstream, but it's not packaged properly yet.
- [`purple-facebook`](https://hydra.nixos.org/build/81611292/log) looks like some python version mismatch?
- [`mikutter`](https://hydra.nixos.org/build/81608564/log) ruby fails on some gtk2 stuff
- [`hardinfo`](https://hydra.nixos.org/build/81613583/log)  fails on some libsoup thing
- [`haskellPackages.gi-soup`](https://hydra.nixos.org/build/81612877/nixlog/1) complains about some libsoup thing too
- `tarball` fails on the ad-hoc update commit it seems

### Not reproducable:
These builds locally for me:
- `erlang`
- `neko`
- `ponyc`
- `julia_10`

### Transient test failures?
- `sage`, looks like transient test failures, but not sure.
- `python3Packages.nilearn` looking at trunk the same test failure happens there sometimes too


The rest looks like transient problems, like eg. `can't allocate more memory` when building `cntk`.

hedning committed:
gnucash: don't fail on glib-2.58's new deprecrations

hedning committed:
purple-facebook: Remove unused mercurial dependency

Having python3 in the PATH and python2's setuptools and enum in PYTHONPATH
breaks any python3 script. Having mercurial in buildInputs makes this
condition true, breaking glib's python scripts, which after 2.58 use python3.

Mercurial isn't actually needed in buildInputs, so removing mercurial is a
simple fix.

jtojnar committed:
gnome-podcasts: init at 0.4.6

hedning commented:
Managed to get to the bottom of the `purple-facebook` problem. The change that unearthed the failure was glib now using python3 for their scripts, in particular `glib-genmarshal`.

Mercurial puts python2's setuptools in the PYTHONPATH which overrides the native `site` module (which seems quite insane). python3 somehow thinks this is perfectly fine until we also add the `enum` python2 package to PYTHONPATH too.

Since mercurial isn't actually used in the build, removing it fixes the problem.

Did not know python2 and 3 interop was this broken and insane, but there we go.

Also pushed a fix for `gnucash`.

jtojnar committed:
gnome3.grilo-plugins: 0.3.7 → 0.3.8

jonafato committed:
gnomeExtensions.icon-hider: 20 -> 21

jonafato committed:
gnomeExtensions.nohotcorner: 17.0 -> 18.0

jonafato committed:
gnomeExtensions.caffeine: unstable-2017-06-21 -> unstable-2018-09-25

hedning commented:
Upstream is releasing 3.30.1 today. In particular `libsoup` is at 2.64.1, in the news file they do note that meson isn't «official» yet so it might be good to switch back to the supported build system in the meantime as we're seeing some failures due to libsoup (eg. with `hardinfo`). Though I haven't tested if it fixes the issue yet.

jtojnar committed:
gnome3.eog: 3.28.3 → 3.28.4

jtojnar committed:
json-glib: 1.4.2 → 1.4.4

hedning commented:
`dbusconfigdir` patch is now upstream for `gdm`: https://github.com/hedning/nixpkgs/commit/1d8d7b7b8b791ac3fd25d0fbd8af930f76413614

I tested `libsoup` without meson, but it didn't fix `hardinfo` or `gi-soup`. The last released `hardinfo` is pretty ancient (though there's some active development at https://github.com/lpereira/hardinfo/

jtojnar committed:
gnome3.libgda: 5.2.4 → 5.2.5

jtojnar committed:
gnome3.tracker: 2.1.4 → 2.1.6

systemd sure service files are no longer installed by default when the build
can't find systemd, so we need to manually specificy the install dir.

jtojnar committed:
gnome3.gnome-remote-desktop: init at 0.1.6

jtojnar committed:
nixos/gnome3: enable remote desktop on wayland

jtojnar commented:
* File-roller does not build https://github.com/mesonbuild/meson/issues/4257
* Tracker does not build https://gitlab.gnome.org/GNOME/tracker/merge_requests/30

dtzWill commented:
While investigating some issue of my own, I went through the commits on the 'gnome-3.30' branch (was looking to see if bumping gdk-pixbuf resolved it, not sure yet!) and noticed a bump to network-manager-applet: 3dfd2fa02ac4ecbf764955a6e8abee91458a0506

Hopefully this is the right place for comments about that--  in an even more happenstance I was going through that project's commits and, well, here: https://gitlab.gnome.org/GNOME/network-manager-applet/compare/1.8.18~10...master (10 commits before the tag through current master)

Particularly the fact that a flurry of commits before the tag about show_all() that apparently needed to be reverted after, having this to say about it: ([linky](https://gitlab.gnome.org/GNOME/network-manager-applet/commit/f3b84748bb3377d503254a387da0f47fddbdcf9c))

> This broke gnome-control-center and nm-connection-editor,
> which call gtk_widget_show_all().
> 
> Two offending gtk_widget_show_all() calls by nm-connection-editor
> were removed in response to the issue for nm-openvpn's certificate
> password ([1], [2], [3], [4]). Of course, older versions of nm-c-e
> and all versions of gnome-control-center are still broken by the change.
> 
> However, there are other issues like missing 802-1x tabs [5].
> 
> It's not acceptable to break existing users of libnma. Even if they
> possibly should be fixed.
> 
> For the moment, revert the patches.

So all of this was a long-winded way of giving you a heads-up -- if nothing else something to keep an eye out for since in case there are problems. :D.

Thanks for working on this!

hedning commented:
`gnome-session-3.30.1` failed the patchPhase. Pushed updated patch (and the glib.bin commit): https://github.com/hedning/nixpkgs/commits/gnome-3.30

worldofpeace commented:
`shotwell` failed build because the `postinstall.py` needed the usual thing.
Also corrected the url for `gnome-notes`.
https://github.com/worldofpeace/nixpkgs/commits/gnome-3.30

dtzWill commented:
NixOS's extraUtils fails with this due to reference that's not allowed-- at first I feared this was the same problem as before (since the symptom is the same as when we were corrupting things re:meson), but I think it's unrelated :).  Regardless I suspect you're running into this and if you haven't you probably will (or have fixed it and I'm on wrong branch :D).

In short the fix is adding `-Dlink-udev-shared=false` to systemd's mesonFlags. I didn't chase down why it suddenly started happening, but I can say I noticed in past few months various other distributions making changes regarding this so perhaps it's due to a meson upgrade they did before us :).

The harder fix, and untested, is that the extraUtils find-libs-and-copy-and-patchelf stuff is both too complicated and in particular fails to account for transitive dependencies (libraries that need other libraries).  Well "fails" may be incorrect, this is possibly a feature and some would argue a case of underlinking but all I mean is this wouldn't happen if the xz dep was pulled in and patched to be used.

I think the solution proposed (meson flags) is the best as it prevents general systemd deps (== take over the world) from being pulled into our early boot process.  We may want to do the same for systemctl (similar option) but that's not necessary so I'll leave that for another time.

Anyway thanks for your work, hope this helps, LMK if you have any questions or whatnot :).

hedning committed:
gxneur: don't fail on new glib deprecration

hedning committed:
hardinfo: fix build with new libsoup

hedning committed:
gitg: 3.26.0 -> 3.30.1

hedning committed:
gnome3.zenity: 3.28.1 -> 3.30.0

hedning commented:
Pushed fixes for `hardinfo` and `gxneur`, and a few updates expressed as squash commits with the correct commit message included in the description. https://github.com/hedning/nixpkgs/commits/gnome-3.30

I also included a patch for meson to fix file-roller, but not sure if it's the best solution?

jtojnar commented:
I have the meson patch in the meson PR

hedning commented:
Didn't actually test `gitg` so it failed as it no longer ships with a `./configure`. So I switched it to meson. Not entirely sure if we need something to replace the make flags or not. It builds and runs fine at least. It seems that upstream is broken on macOS though: https://gitlab.gnome.org/GNOME/gitg/issues/145

hedning commented:
Hmm, so this failure is a bit weird, I'm guessing it's related to linking to `gio-launch-desktop` from `$bin`: https://hydra.nixos.org/build/82891156/nixlog/1:

> different permissions in `/nix/store/sqc7x2x2qq69mr74d73spc6bjh1hjzq3-glib-2.58.1-bin/bin' and `/nix/store/mg2bjvgj4l1mwj9vh9n3aaq68g9h8qbx-glib-2.58.1/bin': 0555 <-> 0555 at /nix/store/9g4wc31j7a2xp22xpgwr0qssfxahxdzl-builder.pl line 75.
collision between `/nix/store/mg2bjvgj4l1mwj9vh9n3aaq68g9h8qbx-glib-2.58.1/bin' and `/nix/store/sqc7x2x2qq69mr74d73spc6bjh1hjzq3-glib-2.58.1-bin/bin'
builder for '/nix/store/9lfhq0bhknq40vvswywi09xyv9jgjdhm-nwjs-env.drv' failed with exit code 2

jonafato committed:
gnomeExtensions.no-title-bar: 8 -> 9

jonafato commented:
@jtojnar I've updated four extensions to versions with 3.30 fixes (icon-hider, nohotcorner, caffeine, and no-title-bar) at https://github.com/jonafato/nixpkgs/commits/gnome-3.30 if you'd like to include them here. Otherwise, I can open up new PRs for them.

hedning commented:
Found the glib problem. When rebasing on top of the `glib.bin` commits it looks like `$out/bin` was never created, so we ended up moving `gio-launch-desktop` to `$out/bin`. Pushed a fixup commit that creates the directory before moving.

jtojnar commented:
Thanks @jonafato and @hedning

hedning commented:
We're unfortunately blocked by #46020 which is somewhat at a standstill. There's a few potential solutions, but we're not really certain if they're actually robust.

cocreature commented:
Thanks for the quick update @hedning!

hedning commented:
Looks like a patch to mutter-3.28 got applied when rebasing so we'll have to remove it: https://github.com/hedning/nixpkgs/commit/15eeb9d655b831a9c17f12239f8cc5a5ff659915

jtojnar commented:
I think you qualify for commit access, could you ask on #nixos-dev?

hedning commented:
Currently failing jobs on `x86_linux` when compared to master: https://hydra.nixos.org/eval/1489224?filter=x86_64-linux&compare=1488047

Some random notes:

- [x] `gnome-keyring` works (just some weird timeout).
- [x] `atlas` builds locally
- [x] pretty sure `cntk` is a random failure (i've seen it fail and a bunch before IIRC)
- [ ] `beets` is pretty strange, reverting 3ce5bbad4e8e2739212796a10bc12f97053c933f fixes the problem for some reason. Specifically `from gi.repository import Gst` fails in some tests.
- [ ] `deepin-dde-api` fails @romildo

jtojnar commented:
```patch
--- /nix/store/x3bvg0rkqail8c8x467mdg3vg9d344vi-python3.6-gst-python-1.14.2/lib/python3.6/site-packages/gi/overrides/Gst.py
+++ /nix/store/yn8wyvy3j5hry11rjiwr3062k0dv1rjf-python3.6-gst-python-1.14.2/lib/python3.6/site-packages/gi/overrides/Gst.py
@@ -566,7 +566,7 @@
                                   time % Gst.SECOND)
 __all__.append('TIME_ARGS')
 
-import _gi_gst
+from . import _gi_gst
 _gi_gst
 
 # maybe more python and less C some day if core turns a bit more introspection
```

hedning commented:
Nice, that seems to be the cause ([which is now upstream](https://gitlab.freedesktop.org/gstreamer/gst-python/commit/af4ade37435fcc31c8489f4d7c7496fef5f74b05)). I think the trigger is having `/nix/store/yn8wyvy3j5hry11rjiwr3062k0dv1rjf-python3.6-gst-python-1.14.2/lib/python3.6/site-packages/` in `PYTHONPATH`.

hedning committed:
gnome3: add hedning to maintainers

hedning commented:
Nice, that fixes the problem

hedning commented:
Needed https://gitlab.freedesktop.org/gstreamer/gst-python/commit/f79ac2d1434d7ba9717f3e943cfdc76e121eb5dd too, as the python2 build failed. should've looked more closely at the commit log before pushing I guess (and actually tested the python2 build).

hedning committed:
deepin.go-gir-generator: fix dde-api with gobject-introspection 1.58+

hedning committed:
gnome3.devhelp: 3.30.0 -> 3.30.1

hedning commented:
I've done a `nix-shell maintainers/scripts/update.nix --argstr path gnome3` and did a bunch of squash commits as upstream is doing a 3.30.2 release these days. Updated spidermonkey too which builds fine locally (tested `gjs` binary).

Currently doing some sanity building. I did encounter a non--reproducable test failure in `gnome-keyring` I'm assuming that's not new?

This hopefully shouldn't cause any more breakage., but if we want to be on the safe side and freeze things I could just push the `deepin.dde-api` fix and tracker update (tracker failed non-deterministically on hydra) in order to hurry things up?

jtojnar commented:
https://github.com/NixOS/nixpkgs/issues/43501?

jtojnar commented:
What’s up with tracker?

hedning commented:
Only one gnome-keyring test failed for me (locally) IIRC, so I think it's something else.

Header wasn't produced in time for tracker: https://gitlab.gnome.org/GNOME/tracker/commit/3cbfaa5b374e615098e60eb4430f108b642ebe76 might be worth to pick the patch (isn't included in 2.1.6).tracker-sparql.h

midchildan committed:
gem-config: Add missing dependencies for gio and gtk2

hedning commented:
Took a look at [newly failed darwin jobs](https://hydra.nixos.org/eval/1489550?filter=x86_64-darwin&compare=1488047&full=#tabs-now-fail) and [dconf](https://hydra.nixos.org/build/84067917/nixlog/1) blocks a lot of jobs. I'm guessing the make fix needs to be ported to meson. [This upstream bug](https://gitlab.gnome.org/GNOME/dconf/issues/47) looks somewhat relevant. @matthewbauer if you're interested in looking at it?

[gst-python](https://hydra.nixos.org/build/84060962/nixlog/3) should be fixed by https://github.com/NixOS/nixpkgs/pull/46310. It might be nice having some assertions that library paths in any generated girs actually exists. Not sure if that's trivial though @jtojnar?

jtojnar commented:
It should be fairly straightforward, find files with `gir` extension, then use xpath to get the `shared-library` attribute of the relevant element. If we want the validation to be robust, we would need to add something like Python and libxml to the closure. That should not be an issue, since they are already pulled in by `gobjectIntrospection`.

hedning commented:
Hmm, we will probably run into trouble with vala generated girs though (vala stuff still depends on gobject-introspection), which would mean fixing that or building a switch. So might be better to tackle it after we get this update done.

`tracker` only install service files by default if it can find systemd at build time. Fixed by explicitly setting the systemd_user_services directory. As far as I can tell systemd is only used to decide if it should install service files.

Pushed stuff to my own branch, not sure how much we're supposed to bog hydra down?

jtojnar commented:
> Hmm, we will probably run into trouble with vala generated girs though (vala stuff still depends on gobject-introspection), which would mean fixing that or building a switch. So might be better to tackle it after we get this update done.

Yes, like every other hook, it should be behind a flag. I think it should be part of #46310. 

> `tracker` only install service files by default if it can find systemd at build time. Fixed by explicitly setting the systemd_user_services directory. As far as I can tell systemd is only used to decide if it should install service files.

Weird, it did install it last time I checked. I even submitted a MR fixing the symlinks. 

> Pushed stuff to my own branch, not sure how much we're supposed to bog hydra down?

We have the quota set so using hydra should be fine. If you cause a significant rebuild shortly after a previous one you can ask on #nixos-dev for cancelling the old one. 

> Edit: found a problem with `polari` too, apparently it now depends on `telepathy-idle` dbus service.

That has always been true.

hedning commented:
Installing polari to systemPackages instead of with nix-env solved the problem for some reason.

jtojnar commented:
It uses `propagatedUserEnvPkgs` hack. We should probably create a module
instead.

On Wed, 14 Nov 2018, 14:59 Tor Hedin Brønner <notifications@github.com
wrote:

> Installing polari to systemPackages instead of with nix-env solved the
> problem for some reason.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/45950#issuecomment-438670872>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AArCY3dYrzcYwvWUyUqJY3Mpo38W1IvBks5uvCG1gaJpZM4WWraN>
> .
>

hedning commented:
The upstream commit that turned off tracker's default systemd service installation in 2.1.6 btw: https://gitlab.gnome.org/GNOME/tracker/commit/9ef964a88d68793b08172eb2aa273875764fa7bf

hedning committed:
gnome3.gnome-characters: add gnome-desktop typelib

hedning commented:
Did a squash and upgrade mutter and gnome shell.

hedning commented:
OK, I would consider this complete for linux, assuming we're happy with the meson update (I think we're pretty much in agreement that cleaning the RPATH is best way to go for now right?). As far as I can tell [all new failures](https://hydra.nixos.org/eval/1489802?filter=x86_64-linux&compare=1488047&full=) (compared to master) are transient. I've run this branch (with my wayland branch on top) as my daily driver for quite a while and haven't found any regressions. `tests.gnome3` and `tests.gnome3-gdm` passes.

There's a few [actual new failures](https://hydra.nixos.org/eval/1489802?filter=x86_64-darwin&compare=1488047&full=) on darwin caused by [`dconf`](https://hydra.nixos.org/build/84067917/nixlog/1) and [`libglade`](https://hydra.nixos.org/build/84011777/nixlog/1). [`gst-python`](https://hydra.nixos.org/build/84060962/nixlog/3) fails too, but should be fixed by #46310. Not sure if any of this should be considered blocking though?

timokau commented:
@NixOS/darwin-maintainers any opinions on @hedning's previous post?

LnL7 commented:
@hedning A lot of those where temporary regressions on master that I fixed last week so I'm not sure. But since gnome is linux only most of this probably doesn't matter.

hedning committed:
gnome3.gnome-desktop: fix path substitution

`substituteAll` intentionally doesn't work with all caps:
https://github.com/NixOS/nixpkgs/issues/28086

jtojnar committed:
vala_0_40: 0.40.6 → 0.40.11

hedning commented:
A [patch](https://github.com/NixOS/nixpkgs/commit/64d50a00991983e90d6a22f39d03b028602e34d6) snuck into meson when rebasing on staging.

jtojnar commented:
Sorry. Could you please push revert patch to staging?

jtojnar committed:
meld: 3.30.2 → 3.30.3

jtojnar committed:
gnome3.gexiv2: 0.10.8 → 0.10.9

jtojnar committed:
gnome3.grilo: 0.3.6 → 0.3.7

jtojnar committed:
nemiver.updateScript: allow unstable versions

jtojnar committed:
empathy.updateScript: allow unstable versions

jtojnar committed:
gnome3.gnome-system-log.updateScript: allow unstable versions

jtojnar committed:
gnome3.gtksourceviewmm.updateScript: allow unstable versions

jtojnar committed:
gtk-vnc.updateScript: allow unstable versions

jtojnar committed:
cantarell-fonts: 0.100 → 0.110

jtojnar committed:
gnome3.updateScript: improve pre-release detection

Previously all versions with minor >= 90 were considered pre-release,
which failed to consider versions like 0.100.

Let’s limit the range to 90–99 and 900–999.

jtojnar commented:
`cantarell-fonts` do not build due to https://github.com/NixOS/nixpkgs/issues/50855

hedning commented:
Seems like we need to fix gettext's envHook as it wrongly adds the cross compile suffix `_FOR_BUILD` to `GETTEXTDATADIRS`, which unfortunately is a huge rebuild :(

hedning commented:
`grilo` upgrade seems to break things, eg. [`grilo-plugins`](https://hydra.nixos.org/build/84536082/nixlog/1):
```
../src/flickr/grl-flickr.c:31:10: fatal error: grilo.h: No such file or directory
 #include <grilo.h>
          ^~~~~~~~~
```

[`enchant`](https://hydra.nixos.org/build/84520323/log) is also breaking due to `hunspell` upgrade (unrelated to our PR though), adding `ncurses` to buildinputs fixes it, but that shouldn't really be necessary:
```
/nix/store/j13c9z0z495wxdc97if8z451x53m1j25-binutils-2.30/bin/ld: cannot find -ltinfo
collect2: error: ld returned 1 exit status
```

hedning committed:
cantarell-fonts: ad-hoc fix for build

`GETTEXTDATADIRS` isn't populated correctly due to empty buildInputs. Borrow from
`GETTEXTDATADIRS_FOR_BUILD` until we fix gettext's envHook.

see: https://github.com/NixOS/nixpkgs/issues/50855

jtojnar commented:
@hedning, try this: https://gitlab.gnome.org/GNOME/grilo/merge_requests/34

hedning commented:
That seems to work :+1: (after changing `enable-gtk-doc` to `gtk_doc`).

hedning commented:
I pushed an update to enchant-1.6 to staging-next btw which fixed the build. meson is now in staging-next so we should probably use that as a base?

hedning commented:
Added your grilo MR. Also added an ad-hoc fix for cantarell-fonts (including the correct output hash) so we'll have fonts until a proper fix is done.

jtojnar commented:
We need to merge to staging, also https://github.com/NixOS/nixpkgs/pull/46310 just broke us.

hedning commented:
I squashed all the gobject-introspection commits and rebased on top of the macos patch (having them as different commits got pretty confusing). I think its correct, building atk works at least, but if you want to take a look before I push the combined commit is here: https://github.com/NixOS/nixpkgs/commit/8cf57ed1858ee1b57717678c3d0e94fb16c5b8bc

hedning committed:
gobjectIntrospection: test our own patches

We patch gobject-introspection somewhat heavily, so add some unit tests that
will catch potential errors when we're eg. updating the patches.

There's two tests:

- Basic test for absolute nix store paths (testing absolute_shlib_path.patch)
- Test for macos `@rpath` sanitizing, including
  fallback_libpath (macos-shared-library.patch)

hedning commented:
OK, pretty certain it's correct. Made some tests: https://github.com/NixOS/nixpkgs/commit/64244fb41b8

jtojnar commented:
Looks great, good work.

hedning commented:
staging-next just got merged, should we just merge to staging now to get onto the next merge-window?

jtojnar commented:
Merged. Many thanks @hedning, @worldofpeace, @grahamc and any other people who contributed.

michaelpj commented:
I don't use gnome, but I've been vaguely following this and I think you all deserve a huge thank you for your tenacious work!
